src/Build.UnitTests/BackEnd/TaskHostFactory_Tests.cs

```diff
@@ -36,8 +36,15 @@ public void TaskNodesDieAfterBuild()
                 string.IsNullOrEmpty(processId).ShouldBeFalse();
                 Int32.TryParse(processId, out int pid).ShouldBeTrue();
                 Process.GetCurrentProcess().Id.ShouldNotBe<int>(pid);
-                Thread.Sleep(1000);
-                Should.Throw<ArgumentException>(() => Process.GetProcessById(pid));
+                try {
+                    Process taskHostNode = Process.GetProcessById(pid);
+                    taskHostNode.WaitForExit(2000);
```

Comment:
Assert that this reported that the process ended (and didn't time out).

Reply:
That's caught by the Should.Throw immediately afterwards. If it times out, `Process.GetProcessById` won't throw, and the test will fail.

Reply:
Not necessarily though, given PID reuse.